packages/runtime-dom/src/helpers/useCssVars.ts

```diff
@@ -57,6 +57,18 @@ function setVarsOnVNode(
   if (vnode.shapeFlag & ShapeFlags.ELEMENT && vnode.el) {
     const style = vnode.el.style
     for (const key in vars) {
+      if (
+        __DEV__ &&
+        vars[key] !== null &&
```

Comment:
it should be if val == null || type isnt string nor number

Reply:
First the tests question: I looked into doing this when I first started but I ran into an issue. If I use the other tests in useCssVars.spec.ts as an example then I cannot pass anything but a string through useCssVars(). The Record in the getter only lists string as a possible value type and if I try anything else I get an error in the IDE and during the test run. If I change the value type from string to any, for instance, then I can create a test expecting a warning easily enough. I don't feel I can just change the Record to suit my testing, so any advice on that would be welcome.

In a Single File Component I can easily pass an invalid value to the style block and the CSS var gets created regardless, but the test environment relying on TypeScript actively prevents an invalid value from being passed. An interesting dilemma to work out.

Second part: I'm not understanding why I would test if val == null since that would be the same result as if val is anything but string or number. Plus if I were to change the && to || after the null test then wouldn't that be "if dev env and null then warn or if any env and not string and not number then warn". I'm not sure I understand the expected result.

From my playing with this feature I can see a use case for wanting to pass a null which is why I would let it through. Consider I have an object of vars that I'm passing along to the style block. I can change any of them to null to turn off the CSS var creation for that var but keep it existing in the object to flip back later. The CSS variable fallback would still work which, to me, would be the point of the null.

Regardless, if I'm understanding the request you would want something like:

`if (
        __DEV__ &&
        vars[key] == null ||
        typeof vars[key] !== 'string' &&
        typeof vars[key] !== 'number'
)`

Is this accurate?